randovania/gui/multiplayer_session_window.py

```diff
@@ -578,6 +582,30 @@ async def duplicate_session(self):
         if new_name is not None:
             await self.game_session_api.duplicate_session(new_name)
 
+    def export_all_presets(self):
+        path = common_qt_lib.prompt_user_for_preset_folder(self)
+
+        if path is None:
+            return
+
+        illegal_chars_pattern = r"^[ .]|[/<>:\"\\|?*]+|[ .]$"
+        world_owners = {world_uid: user for user in self._session.users.values() for world_uid in user.worlds.keys()}
+        world_index = 1
+        world_count = len(self._session.worlds)
+
+        for world in self._session.worlds:
+            owner_name = world_owners[world.id].name if world.id in world_owners else "Unclaimed"
+            preset = world.preset
+            filename = f"World{world_index}-{preset.game.short_name}-{owner_name}-{world.name}-preset.json"
```

Comment:
`...-.{VersionedPreset.file_extension()}`

(aka `rdvpreset`)

Reply:
I didn't even think about that, lol. I wrote all this initially as a once-off thing for BashPrime's async and totally forgot about the file extension